html/common/libs/Auth/main.php

```diff
@@ -377,9 +377,9 @@ function sendMagicLink($email, $redirect) {
         $userID = $DBLIB->getValue("users", "users_userid");
         if (!$userID) return false;
 
-
-        if (!in_array($redirect, $this->VALIDMAGICLINKREDIRECTS)) return false; //Only allow certain redirects, as otherwise this is a vector to spam users with any email text you like
-
+        if (!$CONFIG['DEV']) { //Only allow certain redirects in production
+            if (!in_array($redirect, $this->VALIDMAGICLINKREDIRECTS)) return false; //Only allow certain redirects, as otherwise this is a vector to spam users with any email text you like
```

Comment:
Could this not be `if (!in_array($redirect, $this->VALIDMAGICLINKREDIRECTS) and !$CONFIG['DEV'])`